Clean architecture

mikhailDyoshin opened the pull request:
At last finished building clean architecture for the app.

Commit:
Built an architecture of the app.

Put all business logic from composables into classes.

Commit:
Built clean architecture for the app

The app is diveded into 3 layers: Presentation, Domain and Data. Connected Presentation and Domain layers using MVVM pattern.

dmisuvorov commented on TicTacToeGameCompose/app/src/main/java/com/example/tictactoegamecompose/data/storage/models/Board.kt:
```diff
@@ -0,0 +1,9 @@
+package com.example.tictactoegamecompose.data.storage.models
+
+
+class Board(
```
Обязтельно почитай про data классы в Kotlin, посмотри подходят ли твои модели под них (есть подозрение что да)

dmisuvorov commented on TicTacToeGameCompose/app/src/main/java/com/example/tictactoegamecompose/data/storage/models/Board.kt:
```diff
@@ -0,0 +1,9 @@
+    var logicBoard: List<MutableList<String>>,
+    var crossedCellsCoordinates: MutableList<List<Int>>,
+    var clickedCellsCoordinates: MutableList<List<Int>>,
+    )
```
Слетела разметка (В студии есть автоформатирование класса Ctrl+Shift+L)

dmisuvorov commented on TicTacToeGameCompose/app/src/main/java/com/example/tictactoegamecompose/data/storage/OfflineGameStorage.kt:
```diff
@@ -0,0 +1,422 @@
+        shapeOfAI = "x",
+    )
+
+    @Test
```
Тесты держим в папке test

Commit:
Started to implement dependency injections

Finished to implement DI, have a bug with cells update.

Commit:
Fixed all critical bugs. Finished to implement DI.

It's extremely important to optimise Minimax-algorithm and make it asyncronous.

Commit:
Refactored Minimax algorithm.

Commit:
Optimized Mimimax algorithm.

Optimized Mimimax algorithm with alpha-beta-pruning.

Commit:
Refactored BoardViewModel, deleted CellViewModel

In addition to what the BoardViewModel already did, now it stores a state of each cell on the board and handles all the logic when a cell is clicked.

Commit:
Created previews for the CellScreen composable.

Commit:
Refactored the GameWindowViewModels

Now the GameWindowViewModel is responsable for all game logic. The BoardViewModel is no more.

Commit:
Created a preview for the BoardScreen
